Support cross domain/application copy/paste

AustinMroz opened the pull request:
![AnimateDiff_00001](https://github.com/user-attachments/assets/8ae88dc5-bba8-40c0-9cc2-5e81f579761d)


Browsers place very heavy restrictions on what can be copied and pasted. See:
- https://alexharri.com/blog/clipboard
- https://www.w3.org/TR/clipboard-apis/#mandatory-data-types-x

Further thoughts and TODO
- Should the local storage of copied nodes be removed entirely?
  - It's unlikely a person would want to paste nodes from several days back, but having multiple items in the clipboard may be desirable. Should "copy node -> copy text -> paste text -> paste node" be possible?
- Need to add extra safety to the checks instead of a try/catch block

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6087-Experimental-cross-domain-application-copy-paste-28e6d73d36508154a0a8deeb392f43a4) by [Unito](https://www.unito.io)

AustinMroz committed:
Experimental cross domain/application copy/paste

AustinMroz committed:
Fix change tracking. Loosen target restrictions

AustinMroz committed:
Further refine target restrictions

The instanceof check is required for the type inspection to be valid,
but in general, if text is being edited (such as the login password
field), the text input should always have priority

AustinMroz committed:
Improve selector, loosen copy restrictions

AustinMroz committed:
Move to function, satisfy typechecking

christian-byrne commented on src/composables/useCopy.ts:
```diff
@@ -9,28 +9,24 @@ export const useCopy = () => {
     if (
-      (e.target instanceof HTMLTextAreaElement &&
-        e.target.type === 'textarea') ||
-      (e.target instanceof HTMLInputElement && e.target.type === 'text')
```
Did we want to return early for all `HTMLInputElement` including things like radio, checkboxes, button?

AustinMroz replied:
Very good point. From a quick pass of all input element types, looks like we'd want.  
- Handled by canvas: button, checkbox, file, hidden, image, radio, range, reset, search, submit
- Handled by element: color, date, datetime-local, email, month, number, password, search, tel, text, time, url, week, datetime

Probably best implemented as a whitelist of input element types we allow the canvas to handle for future-proofing?

AustinMroz replied:
And also probably best to revert changes here to be *just* clipboard support and have any targeting changes be handled over in #6055

AustinMroz replied:
Added whitelist here as suggested. For an easily testable places this matters: Try pasting after toggling the value of a boolean in vue mode.

AustinMroz committed:
Replace HTML parsing with regex

AustinMroz committed:
Add a whitelist of allowed input types

and moved the check into a utility file so that the check is shared
between the copy and paste code.

christian-byrne reviewed (approved):
LGTM
